### PR TITLE
Oauth middelware

### PR DESCRIPTION
**Pull request by JesseNaidoo**

### CODE REVIEW

**Review by tristanphips — APPROVED** [review]

File: api/controllers/account.controller.js [hunk 1]
```diff
@@ -108,4 +109,29 @@ export const checkIfAccountExists = (account_id) => {
...
+    let accountObject = null;
+    await DbUtils.spyderdb.oneOrNone('SELECT * FROM account WHERE uid = $1', [uid])
+        .then(async data => {
+            if (data) {
+                accountObject = {
+                    username: data.username,
+                    img_url: data.img_url,
+                }
+            } else {
+                await addUser(uid, null);
```
Comment (ryanbasiltrickett) [inline api/controllers/account.controller.js:124]: Add image null

File: api/controllers/account.controller.js [hunk 2]
```diff
@@ -2,18 +2,19 @@ import {HttpStatusCodes, DbUtils} from '../utils/index.js';
 
 export const createAccount = async (req, res) => {
```
Comment (ryanbasiltrickett) [inline api/controllers/account.controller.js:3]: Remove create account

File: api/controllers/auth.controller.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import {AccountController} from './index.js'
+
+
+export const login = async (req, res) => {
+    try {
+        const {uid} = req.body;
+        if (!uid) {
+            res.status(HttpStatusCodes.BadRequest).send({message: "Invalid payload"});
+        } else {
+            const username = await AccountController.checkIfUIDExists(uid);
```
Comment (ryanbasiltrickett) [inline api/controllers/auth.controller.js:11]: Change var name

File: api/controllers/auth.controller.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            res.status(HttpStatusCodes.OK).send({username: username});
```
Comment (ryanbasiltrickett) [inline api/controllers/auth.controller.js:12]: Pass object to send

**Review by ryanbasiltrickett — CHANGES_REQUESTED** [review]
Please change these issues

**Pull request merged**